test/e2e/karmadactl_test.go

```diff
@@ -0,0 +1,330 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl"
+	"github.com/karmada-io/karmada/pkg/karmadactl/options"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("Karmadactl promote testing", func() {
+	var karmadaConfig karmadactl.KarmadaConfig
+	var member1 string
+	var member1Client kubernetes.Interface
+
+	ginkgo.BeforeEach(func() {
+		karmadaConfig = karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+		member1 = framework.ClusterNames()[0]
+		member1Client = framework.GetClusterClient(member1)
+	})
+
+	// test namespaced resource deployment with cluster resource namespace
+	ginkgo.Context("Test promoting namespaced resource: deployment", func() {
+		var deployment *appsv1.Deployment
+		var deploymentNamespace, deploymentName string
+
+		ginkgo.BeforeEach(func() {
+			karmadaConfig = karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+
+			deploymentNamespace = fmt.Sprintf("karmadatest-%s", rand.String(RandomStrLength))
+			deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+			deployment = helper.NewDeployment(deploymentNamespace, deploymentName)
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveDeployment(kubeClient, deploymentNamespace, deploymentName)
+			framework.RemovePropagationPolicy(karmadaClient, deploymentNamespace, deploymentName+"-propagation")
+			framework.RemoveNamespace(kubeClient, deploymentNamespace)
+		})
+
+		ginkgo.It("Test promoting a deployment from cluster member", func() {
+
+			//  Step 1,  create ns and deploy on cluster member1
```

Comment:
We usually add a space at the beginning of the comment, and the same goes for other places.